NEXT Figma Documentation and Guide

mmailaender opened the pull request:
## Description

Adjust the Figma Landing page + add Tutorials to the docs to prepare for the v3 launch.

Progress
- [x] Figma Landing page
- Tutorials
  - [x] Chapter 1: Set-up the Figma UI Kit
  - [x] Chapter 2: How to use the Figma Plugin
  - [x] Chapter 3: How to update the Figma UI Kit

Figma UI Kit for testing: https://discord.com/channels/1003691521280856084/1217864742643957832/1303390206946316310

--- 
### Updates

- We've changed the` .png` images to `.webp` to have smaller and more performant images while keeping the transparent background to provide some semi-dark/light image support (We will later provide dark and light images, but for now, we're keeping it lean).
- We're sticking with numbering in the file names, as the files are adjusted individually for the tutorials, and the numbering increases clarity.

---
The following tutorials are not MVP relevant and will be delivered afterwards:
  - Chapter 3: Wireframes
  - Chapter 4: Component
  - Chapter 5: Colors 
  - Chapter 6: Typography
  - Chapter 7: Slots and Properties

## Checklist

Please read and apply all [contribution requirements](https://next.skeleton.dev/docs/resources/contribute).

- [x] Your branch should be prefixed with: `docs/`, `feature/`, `chore/`, `bugfix/`
- [x] Skeleton v3 contributions must target the `next` branch (NEVER `dev` or `master`)
- [x] Documentation should be updated to describe all relevant changes
- [x] Run `pnpm check` in the root of the monorepo
- [x] Run `pnpm format` in the root of the monorepo
- [x] Run `pnpm lint` in the root of the monorepo
- [x] Run `pnpm test` in the root of the monorepo
- [x] If you modify `/package` projects, please supply a Changeset

## Changsets

[View our documentation](https://next.skeleton.dev/docs/resources/contribute/get-started#changesets) to learn more about [Changesets](https://github.com/changesets/changesets). To create a Changeset:

1. Navigate to the root of the monorepo in your terminal
2. Run `pnpm changeset` and follow the prompts
3. Commit and push the changeset before flagging your PR review for review.

mmailaender commented:
We've changed the` .png` images to `.webp` to have smaller and more performant images while keeping the transparent background to provide some semi-dark/light image support (We will later provide dark and light images, but for now, we're keeping it lean).

mmailaender commented:
Pull request is ready for review. 
Figma UI Kit for testing: https://discord.com/channels/1003691521280856084/1217864742643957832/1303390206946316310

endigo9740 commented on sites/next.skeleton.dev/src/content/docs/get-started/figma.mdx:
```diff
@@ -1,26 +1,57 @@
+import ImgPlugin from '@images/get-started/figma/plugin.webp';
+import ImgTutorial from '@images/get-started/figma/tutorials.webp';
+
+<div class="relative">
```
The first thing I want to know as a user is what this page all about. For example, see the large prominent text on the [Layouts page](https://next.skeleton.dev/docs/get-started/layouts). It would be great to have something like this. A quick suggestion:

> "Skeleton Labs has coordinated with professional designers to provide a premium Figma UI Kit. This implements all available Skeleton features and components, while also providing a dedicated theme import feature to help your designers get started quickly."

(feel free to improve as desired)

endigo9740 commented on sites/next.skeleton.dev/src/content/docs/get-started/figma.mdx:
```diff
@@ -1,26 +1,57 @@
+
+## Make it your own
+
+Import your custom Skeleton Theme with our Figma plugin and let it shine in your design.
```
I'd suggest:

> Skeleton theme (lowercase)

endigo9740 commented on sites/next.skeleton.dev/src/content/docs/get-started/figma.mdx:
```diff
@@ -1,26 +1,57 @@
+
+## Pair Figma with Skeleton.
+
+From setting-up Skeleton in Figma to creating wireframes and pixel perfect design, we cover your back.
```
This lines reads a bit odd, so I'd suggest:

> From the initial setup, to creating wireframes and pixel-perfect designs, we've got your back.

endigo9740 commented on sites/next.skeleton.dev/src/content/docs/get-started/figma.mdx:
```diff
@@ -1,26 +1,57 @@
 ## Get Access
 
-> Maintainer Note: this will be made available alongside the stable release of Skeleton v3.
+Support the development of Skeleton by purchasing the Figma UI Kit! Your purchase helps ensure continued development and success of Skeleton as free and open-source project.
```
I'm a little leery of how this information is presented. Obviously the Figma designs now longer directly benefit Skeleton. I might suggest leaving that more understate, if not removing the line altogether. Less is more some times.

endigo9740 commented on sites/next.skeleton.dev/src/content/docs/tutorials/figma/chapter-1.mdx:
```diff
@@ -0,0 +1,110 @@
+
+        <div class="card-enhanced border-[1px] border-surface-200-800 space-y-2 p-4 md:p-8 flex flex-row justify-between items-center">
+            <div className="flex flex-row gap-4 items-center">
+            <Image src={SvgSkeleton} class="rounded-container" alt="Skeleton" />
```
The SVG image here gets a bit lost. It's very detailed for it's size. Maybe try scaling that back to just a simple document image with the Skull and Figma icons?

endigo9740 commented on sites/next.skeleton.dev/src/components/docs/FigmaBuyNow.astro:
```diff
@@ -0,0 +1,82 @@
+				<p>Figma Skeleton v3 Components library</p>
+			</div>
+			<div class="flex flex-row items-center gap-2">
+				<div class="flex w-5 h-5 preset-tonal-primary rounded-full justify-center items-center">
```
Make sure to change CTA buttons to `preset-filled`

endigo9740 commented on sites/next.skeleton.dev/src/content/docs/tutorials/figma/chapter-1.mdx:
```diff
@@ -0,0 +1,110 @@
+                <p>Complete set of themed components for Skeleton v3</p>
+            </div>
+            </div>
+            <a href="https://polar.sh/skeleton-kit/" type="button" class="btn preset-filled-primary-500" target="_blank">Purchase</a>
```
Make sure to change the CTA button to `preset-filled`

endigo9740 commented on sites/next.skeleton.dev/src/content/docs/tutorials/figma/chapter-1.mdx:
```diff
@@ -0,0 +1,110 @@
+    <ProcessStep step="2">
+        ### Create a New Project
+
+        In Figma, go to `Home`¹ and select `All projects` from the left sidebar². On the right side, click `+ Project`³ and name the project `Libraries`⁴.
```
I feel like we're way overusing the `code` highlighting, which feels very busy and a bit harder to read. Basically: "if everything is important, nothing is". So I'd suggest trying to strip this down to simpler and more straight forward text. Maybe use italics instead?

Additionally the super/sub text should probably be markup rather than unicode characters. This isn't possible in markdown, but MDX supports HTML, which should enable:

* `<sup></sup>`
* `<sub></sub>`

---

Alternatively, I think the small numerals may get lost with certain fonts and font sizes from various themes. Maybe consider rewriting using bulleted lists:

```
Follow these steps in Figma:

1. Go to the Home icon (top-left).
2. Select the All Projects option from the left sidebar.
3. Select Project from the right-side of the page.
4. Provide your project name in the dialog prompt.
```

endigo9740 commented on sites/next.skeleton.dev/src/content/docs/tutorials/figma/chapter-2.mdx:
```diff
@@ -0,0 +1,150 @@
+
+        Return to the `Skeleton Themes` plugin in Figma, paste the theme code⁹, and click `Upload`¹⁰.
+
+        **Note:** Uploading the theme may take 3–5 minutes, depending on your device.
```
I've been putting small tips and notes like this in blockquotes:

```
> This is an example blockquote.
```

endigo9740 commented on sites/next.skeleton.dev/src/content/docs/tutorials/figma/chapter-2.mdx:
```diff
@@ -0,0 +1,150 @@
+
+---
+
+## Import Custom Theme in Project
```
I'm not immediately clear the difference between:

- Import Custom Theme
- Import Custom Theme in Project

Do I as a user go through both sets? Is there a condition where I would use one versus the other? Probably best to make this more clear.

mmailaender replied:
Typo: Should be `Apply Custom Theme in Project`

endigo9740 commented on sites/next.skeleton.dev/src/content/docs/tutorials/figma/chapter-2.mdx:
```diff
@@ -0,0 +1,150 @@
+
+## Import Custom Theme in Project
+
+### Prerequisites:
```
Given we're going to have so many chapters and this is meant to be read in a linear fashion, the requisite list feels redundant. If a step is optional, then indicate it plainly upfront. We do this with titles sometimes too so it's visible in the Table of Contents:

```
1. Do Something
2. Optional: Do Something Else
3. Wrapping up
```

mmailaender replied:
Most users do not follow orders (although it would make things so much easier), so we have the prerequisites at the top indicating chapters that are a MUST before you continue to remind all the people who skipped directly to this chapter. You will see in the upcoming tutorials that
1. only a limited amount of chapters are a hard prerequisite
2. we have implemented a fallback system to cover people who jump directly to e.g. Chapter 4 without reading e.g. Chapter 3.

So stay tuned :)

endigo9740 reviewed (changes requested):
@mmailaender Again this may look like a lot, but it's a lot of little things. If you have any questions about this feedback please let me know here or in Discord.

I have to say, everything is looking great! Tons of information and screenshots is always nice. I'm sure users will really appreciate this!

mmailaender commented on sites/next.skeleton.dev/src/content/docs/tutorials/figma/chapter-2.mdx:
```diff
@@ -0,0 +1,150 @@
+    <ProcessStep step="1">
+        ### Open Skeleton v3 Figma Library
+
+        **Important:** The plugin is compatible only with the Skeleton v3 Figma Library.
```
The answer to your question lies one step afterward. I moved the disclaimer to this step.

mmailaender commented on sites/next.skeleton.dev/src/content/docs/get-started/figma.mdx:
```diff
@@ -1,26 +1,57 @@
+<div class="relative">
+	<Image src={ImgHero} class="rounded-container" alt="hero" />
+	<a
+		class="btn btn-lg preset-filled-primary-500 flex items-center gap-2 whitespace-nowrap absolute top-6 right-6 shadow-lg w-48"
```
That is interesting. What are your thoughts over having a neutral style button as a CTA? 
The CTA is the primary goal what the user should do on the page. In our case it's buying the Figma assets. So having a neutral style button (White and black) leads to no hierarchy differentiation between the CTA and the surrounding content. 
Companies that have optimized their CTA color contrast have seen substantial increases in conversion rates:
- Shopify reported a 12% increase in conversions after adjusting their CTA buttons for higher contrast.
- HubSpot experienced a 20% boost in user interactions by using bold orange against a white background.
- Booking.com improved accessibility and conversion rates by 15% through contrasting blue buttons on a white background

endigo9740 replied:
![Screenshot 2024-11-13 at 9 48 15 AM](https://github.com/user-attachments/assets/8e7e8ffd-1471-4f3a-99cd-7affb67d886a)

It's not an issue of contrast here. It's an issue of assigning meaning. Neutral doesn't mean low contrast, in this case the "Get Started" button is still clearly visible, right? But we reserve color for even more critical items on a level above that. Such as the announcement, various alerts, etc.

It's similar to the comment I made the other day, we can't make every single action on the page the "most important" on the page. Otherwise nothing is important. Instead, think in terms of levels or grades.

- Most critical items - colored and really catch your eye
- Positive actions - ala CTAs, neutral, but still high contrast
- Neutral actions - the least contrast, using outlined currently

endigo9740 replied:
From a design perspective we're also trying to keep a cohesive design. We need the Figma section to match the aesthetic of the rest of the app. If we decide to change the aesthetic, we'll do it across the board. Not at a one-off.

mmailaender replied:
Contrast is not only 100 vs 900. It can be also black-900 vs blue-500, and that's what I'm talking about. The most important objective on the Figma page is the conversion CTA. The whole page works towards that direction. Everything else is secondary.

endigo9740 replied:
I do believe the Figma page is already fairly unique. Lots of big images, the "buy now" panel looks great, and I don't think anyone will have any trouble locating how to purchase if they are interested. But I'm happy to work with you on that once I have more bandwidth.

I have at least one more cosmetic pass in mind before launch. The homepage especially, which is currently a placeholder. We'll have something much more detailed, much informative and interactive, not unlike the v2 site (but hopefully even better). That can also include an additional call to action pointing at the Figma section.

mmailaender commented:
Ready for another round of review @endigo9740 ✌️

1. We've reworked the whole landing page and implemented a conversion journey. We start with the Why (product's purpose and value proposition), transition to How (we solve the user's problem), and what (is the concrete product), a social proof section, and then finish off with the pricing and final CTA. Ideally, this page would be its own landing page as it conflicts a bit with the structure of the docs (as we also discussed), but for now, we have fitted it in to have a starting solution.
2. Addressed or fixed open comments

endigo9740 commented:
Hey @mmailaender @bohdank1995, so after doing a bit of review of this I have a couple ideas I want to run by you...

## Per the Promo Page

First, regarding the core Figma page `/docs/get-started/figma`. I can see now what you guys are going for with this - it's more of a promotional or landing page that discusses the benefits of the Figma Kit and works to convert folks to paying users.

That totally makes sense, but it's a bit outside the original intention. Originally I just this to be another documentation page that work would push users to to a prom/landing page, but more standalone. Be that on Polar, etc. So, what I'm proposing is we move away from the MDX-based content page, and instead implement a new top-level page.

This would look like this:

- Temporarily disable `/src/content/get-started/figma` by renaming `to `_figma`. As underscore routes are ignored. 
- Create a new page in `/src/pages/figma/index.astro`. Astro pages fully support HTML, just like components.
- We then rework the navigation to include Figma as a top-level link in the header (Docs | News | Themes | Figma)
- Once everything is moved, you are then free to delete the `/content` version

Just make sure to include the wrapping `<LayoutRoot>` component like the 404 and other pages.

This then gives you guys full reign over the page structure and HTML for the design, much like how we're handling [News](http://localhost:4321/news). Plus you can still insert the existing `<FigmaBuyNow>` and `<NavGrid>` components as you see fit. The only things you will lose is the left/right columns for navigation and table of contents. And image paths may need to be modified a bit. But I think this should be more in the ballpark of what you seem to be after.

## Per Tutorials

Additionally, I think we do keep the tutorials in MDX/documentation format, but we can go ahead and implement a new Top-Level category in the left hand sidebar called Tutorials. Think something like this:

```
Get Started
- Introduction
- ...
- Tailwind Guide

Tutorials <--- New top level section
- Dark Mode
- Layouts
- Figma UI Kit <--- Figma tutorials live here

Design
- Themes
- ...
```

That way discovery comes from either the top level link, or from some browsing to the tutorial. The promo page and tutorial should likely feature a CTA for quick navigation to and from.

## Upcoming Design Refresh

Finally, as for the promo page design - feel free to be creative, but know I do have plans to revise the homepage just prior to launch to something more intricate and informative. We should ideally keep these in sync so everything looks uniform. Unfortunately I likely won't have a chance to implement those revisions until after the 1st of the year, leading up to the full launch. So maybe keep you design conservative for now, to keep it nimble. Otherwise we can follow up with a revision asap.

## My Help

Finally, note I could probably whip up most of these structural changes in a couple hours. If you are down for this let me know and I'll try to implement right away. Either way, just let me know what you think on Discord and we'll go from there!

endigo9740 commented:
@micha.mailaender @bohdank1995 

I called this out on Discord, but wanted to repeat it here.

---

One thing I did notice. We've now moved the Figma tutorials section under `/docs/guides/figma`:
https://skeleton-next-krbzc29ja-skeleton-labs.vercel.app/docs/guides/figma

But it looks like the tutorial pages themselves are still under `/docs/tutorials/figma/chapter-1`
https://skeleton-next-krbzc29ja-skeleton-labs.vercel.app/docs/tutorials/figma/chapter-1

All three chapters will need to be moved under `/docs/guides/figma/chapter-{number}`. This will help ensure the Breadcrumb links are functional.

endigo9740 commented on sites/next.skeleton.dev/src/pages/figma/index.astro:
```diff
@@ -5,16 +5,157 @@ import LayoutRoot from '@layouts/LayoutRoot.astro';
+				<div class="flex flex-col gap-4 basis-1/2">
+					<h5 class="h5">Align quickly with prebuilt Skeleton components.</h5>
+					<p class="text-surface-800-200">
+						All components from the Skeleton design system are prebuilt and ready to use in Figma, saving you time and ensuring consistency.
```
Suggested text:

> All components from Skeleton have been recreated in Figma, saving you time and helping establish consistency.

endigo9740 commented on sites/next.skeleton.dev/src/pages/figma/index.astro:
```diff
@@ -5,16 +5,157 @@ import LayoutRoot from '@layouts/LayoutRoot.astro';
+				<div class="flex flex-col gap-2 basis-1/2">
+					<h5 class="h5">Iterate faster with tutorials designed for every step of the design process.</h5>
+					<p class="text-surface-800-200">Learn how to:</p>
+					<ul class="list-inside list-disc space-y-2 text-surface-800-200">
```
The second bullet point is long enough that it's wrapping. We can fix alignment with CSS, but for the sake of time I might suggest trying to shorten the text a bit.

![Screenshot 2024-12-16 at 2 19 26 PM](https://github.com/user-attachments/assets/d620e5f9-5152-4c4d-b662-b5dacf334e75)

This will appease my OCD :)

mmailaender replied:
It's responsive and only appears if the width is between 768 - 907 and below 458px. For all other sizes, it does not wrap. In my opinion, this is fine.

endigo9740 commented on sites/next.skeleton.dev/src/pages/figma/index.astro:
```diff
@@ -5,16 +5,157 @@ import LayoutRoot from '@layouts/LayoutRoot.astro';
+			</blockquote>
+
+			<div class="flex flex-col gap-12 pb-12">
+				<a class="text-surface-700-300">Rob, in Skeleton community</a>
```
Love the quote, but I might suggest:

> Rob, Skeleton community member

endigo9740 commented on sites/next.skeleton.dev/src/pages/figma/index.astro:
```diff
@@ -5,16 +5,157 @@ import LayoutRoot from '@layouts/LayoutRoot.astro';
+
+			<div class="flex flex-col gap-12 pb-12">
+				<a class="text-surface-700-300">Rob, in Skeleton community</a>
+				<a class="btn btn-lg preset-filled-primary-500 flex items-center gap-2 whitespace-nowrap shadow-lg w-fit" href="#get-access">
```
I appreciate the call out, but I don't think this is needed. We have the Discord link in the page header and footer for every page. I plan to introduce a dedicated section for this in the homepage as well - perhaps pulling real numbers of I can find the appropriate APIs.

mmailaender replied:
That's not a callout for Discord :) It's one of the conversion buttons that moves you to the buy section showing how many people use the Skeleton Figma Kit (v2). Everything on that page works toward the goal of converting the lead :)

endigo9740 commented on sites/next.skeleton.dev/src/components/docs/FigmaBuyNow.astro:
```diff
@@ -0,0 +1,113 @@
+
+<div class="flex flex-col bg-transparent rounded-container border-[1px] border-surface-200-800 shadow-lg overflow-hidden">
+	<div class="flex flex-col p-6 w-full">
+		<span class="text-xl font-semibold h-full pb-6">Figma UI Kit for Skeleton includes:</span>
```
Recommended text:

> Figma designs for all available components

We only refer to things as "v3" for now while we're between the v2/v3 updates.

endigo9740 commented on sites/next.skeleton.dev/src/pages/figma/index.astro:
```diff
@@ -5,16 +5,157 @@ import LayoutRoot from '@layouts/LayoutRoot.astro';
+
+		<h2 class="h2 pb-8" id="get-access">Get Access</h2>
+
+		<FigmaBuyNow />
```
We need some kind of contact information below the Buy Now component. If users are having trouble, we don't want our contributors or community members being forced to deal with this. They should have a direct link of communication with you guys.

This can take one of more forms here:

- A direct support email list (add an anchor with a `mailto:` URL)
- A company website with your contact information
- We can also setup a `#figma-ui-kit` channel under Skeleton Labs on Discord.

I'd recommend something like this:

> The Skeleton Figma UI Kit is created with ❤️ from [Micha](https://www.linkedin.com/in/micha-mail%C3%A4nder-548aba16a/), [Bohdan](https://www.linkedin.com/in/bohdan-karpliak-659a41162/), and [Skeleton Labs](https://www.skeletonlabs.co/). If you have questions about your purchase or need support, please contact us at  [email@email.com](mailto:email@email.com).

If you guys don't have an email handy to use, then we can use `admin@skeleton.dev` temporarily. But I'd prefer you guys managed this asap.

mmailaender replied:
I like the suggestion with the #figma-ui-kit so we have one source for the community, and it has a search function, making it easier to deal with repeating questions 👍

endigo9740 replied:
Note the `#figma-kit-support` channel has been added on Discord, but will remain private until the Kit launches.

endigo9740 commented on sites/next.skeleton.dev/src/content/docs/guides/figma.mdx:
```diff
@@ -0,0 +1,27 @@
+
+Explore step-by-step guides to unlock the full potential of Skeleton in your design workflow. Whether you're setting up for the first time or mastering advanced features, these tutorials will guide you every step of the way.
+
+Discover all the benefits of <a href='/figma' className='anchor'>Skeleton Figma UI library. </a>
```
The callout is getting lost here. I might suggest you utilize one of these large button panels like shown here:

https://next.skeleton.dev/docs/get-started/plugin#base

Make the CTA something like "Get the Figma UI Kit".

![Screenshot 2024-12-16 at 2 40 55 PM](https://github.com/user-attachments/assets/5eee141a-2dc3-409a-a8da-e8b605001db4)

endigo9740 commented on sites/next.skeleton.dev/src/content/docs/guides/figma.mdx:
```diff
@@ -0,0 +1,27 @@
+import { Shapes } from 'lucide-react';
+import { SwatchBook } from 'lucide-react';
+
+Explore step-by-step guides to unlock the full potential of Skeleton in your design workflow. Whether you're setting up for the first time or mastering advanced features, these tutorials will guide you every step of the way.
```
Feel free to make this the larger prominent text as seen on most other doc pages.

Example:
https://skeleton-next-krbzc29ja-skeleton-labs.vercel.app/docs/get-started/fundamentals

endigo9740 reviewed (changes requested):
A few more things here. Most of these should be quick fixes.

Sarenor commented on sites/next.skeleton.dev/src/pages/figma/index.astro:
```diff
@@ -5,16 +5,157 @@ import LayoutRoot from '@layouts/LayoutRoot.astro';
+		<div class="flex flex-col gap-6 py-12">
+			<blockquote class="blockquote text-3xl not-italic">
+				"...the Figma is very well put together and the most robust that I could find out of all the UI kits so our company chose to go with
+				Skeleton 🙂
```
If you open a quote, you should probably close it, too ;) `"`

mmailaender replied:
Good catch - thx 👍
